### ISSUE TITLE
Vorta needs to be launched several times before it works correctly

### ISSUE THREAD

**Issue by luckyrat** [borgbase/vorta#405]
I had a lot of trouble setting up Vorta on a fresh Ubuntu 19.10 installation recently. I'm not sure if it's due to a single problem or multiple ones but I see at least some similarities with #318 .

Here's the first command I executed and the output...

```
$ flatpak run com.borgbase.Vorta

STDOUT:

QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Gtk-Message: 12:25:36.431: Failed to load module "canberra-gtk-module"
Gtk-Message: 12:25:36.432: Failed to load module "canberra-gtk-module"
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Qt: Session management error: None of the authentication protocols specified are supported
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Qt: Session management error: None of the authentication protocols specified are supported
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Qt: Session management error: None of the authentication protocols specified are supported
failed to register service "org.kde.StatusNotifierItem-4-5"
```

After seeing no sign of Vorta, neither the main window nor the tray icon, I did Ctrl-C to close it down and executed it again. As evidenced below, that did not work either.

```
$ flatpak run com.borgbase.Vorta

STDOUT:

QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Gtk-Message: 12:30:00.790: Failed to load module "canberra-gtk-module"
Gtk-Message: 12:30:00.790: Failed to load module "canberra-gtk-module"
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Qt: Session management error: None of the authentication protocols specified are supported
An instance of Vorta is already running. Opening main window.
```

At this point the main window appeared as an active task in the Ubuntu task bar and I could interact with it successfully but the tray icon was still absent. After finding and force killing a process (`bwrap --args 35 vorta`) I ran it for a 3rd time and it worked as expected:

```
$ flatpak run com.borgbase.Vorta

STDOUT:

QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Gtk-Message: 12:31:28.806: Failed to load module "canberra-gtk-module"
Gtk-Message: 12:31:28.806: Failed to load module "canberra-gtk-module"
QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700
Qt: Session management error: None of the authentication protocols specified are supported
```

The same triple launch is required when Vorta is first invoked from the Ubuntu shell rather than a terminal, although perhaps a system restart after the first terminal triple-launch would have picked up some environment changes and made this un-necessary. Still, whether it takes 6 or 3 launches to make it work, I think this is not working quite as intended :-)

After a system restart, Vorta does not start automatically. When it is manually started the system tray icon is visible, scheduled tasks execute correctly and the GUI can be successfully invoked from the system tray.

So it looks as though my immediate problems have been worked around, at least until I next have to install Vorta on a fresh Linux system. I suspect the lack of auto-start when the system starts is a separate issue that I'll try to resolve at a later date (I never got this to work on my previous installation on Ubuntu 18.04 either but I think that was not installed via flatpak).

I think I've included everything that I can recall but if you have any questions that might help you to streamline the installation process in future, just let me know and I'll try my best to provide extra information.

**Comment by Hofer-Julian** [comment]
Unfortunately, i couldn't reproduce these issues on a fresh Ubuntu 19.10 install in a virtual machine.
I installed flatpak via apt, installed Vorta, rebooted and then started it.
It works without problems as far as I can see, including autostart.

**Comment by luckyrat** [comment]
I didn't reboot in between installing Vorta and starting it. Perhaps that's
the difference?

**Comment by Hofer-Julian** [comment]
I don't think so, after installing Vorta, I noticed that it didn't start before a reboot, but should me instead that a reboot was needed. So I think I did the exact same you did.

I don't think we can reproduce your error, but I am curios, does autostart still not work?

**Comment by luckyrat** [comment]
> but should me instead that a reboot was needed.

Sorry but I'm not following this part of the sentence. Did I miss a warning
or instruction somewhere that said a reboot was a critical step?

> does autostart still not work?

Correct. It still does not work. I have no prior experience with flatpak
but if you're able to point me to some logs (or instructions to enable
autostart debug logs) I'm happy to forward on the relevant bits next time I
restart the system.

**Comment by Hofer-Julian** [comment]
Should - > told, autocorrect...

It would be helpful if you deactivate and then activate autostart again

**Comment by luckyrat** [comment]
Ah, great! That's fixed the autostart problem. I did read an issue before installing Vorta which stated that disabling and re-enabling the autostart option is required after installation so I had already performed this workaround before the first system restart after I'd got Vorta to start. It appears that I must have done it too early during the installation procedure since doing it for a 2nd time has now enabled autostart.

So it sounds as though there was a notice somewhere which told me I must restart my system in between installing Vorta and running it for the first time? You won't be surprised to hear that I didn't see this warning but whether that's down to a lack of concentration on my part or something in the Vorta or Flatpak installation process that didn't work quite right, I'll leave to you experts and/or future feedback from other users to determine :-)

Or perhaps this is just the normal procedure for every application installed via Flatpak? Vorta is my first and only use of that packaging system so I may just have not realised that in this regard Flatpak installs behave differently to a typical application install on Linux.

If it's something that is under your control then changing the procedure so that a restart is not required would obviously eliminate this source of confusion entirely but if it's not feasible to change that, or make a more visible warning to this effect, at least this issue can now help others to understand the situation so I'm fine with you closing/resolving the issue however you see fit.

Thanks for taking the time to investigate.
